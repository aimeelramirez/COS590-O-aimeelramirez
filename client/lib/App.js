Object.defineProperty(exports,"__esModule",{value:true});exports.default=App;var React=_interopRequireWildcard(require("react"));var _native=require("@react-navigation/native");var _bottomTabs=require("@react-navigation/bottom-tabs");var _HomeScreen=require("./screens/HomeScreen");var _reactNative=require("react-native");var _jsxRuntime=require("react/jsx-runtime");var _this=this,_jsxFileName="C:\\Users\\aimee\\OneDrive\\Desktop\\Projects2024\\COS590-0-aimeelramirez\\client\\src\\App.jsx";function _getRequireWildcardCache(e){if("function"!=typeof WeakMap)return null;var r=new WeakMap(),t=new WeakMap();return(_getRequireWildcardCache=function _getRequireWildcardCache(e){return e?t:r;})(e);}function _interopRequireWildcard(e,r){if(!r&&e&&e.__esModule)return e;if(null===e||"object"!=typeof e&&"function"!=typeof e)return{default:e};var t=_getRequireWildcardCache(r);if(t&&t.has(e))return t.get(e);var n={__proto__:null},a=Object.defineProperty&&Object.getOwnPropertyDescriptor;for(var u in e)if("default"!==u&&Object.prototype.hasOwnProperty.call(e,u)){var i=a?Object.getOwnPropertyDescriptor(e,u):null;i&&(i.get||i.set)?Object.defineProperty(n,u,i):n[u]=e[u];}return n.default=e,t&&t.set(e,n),n;}var Tab=(0,_bottomTabs.createBottomTabNavigator)();var AppNavigator=function AppNavigator(){return(0,_jsxRuntime.jsx)(_reactNative.View,{style:{backgroundColor:'pink',flex:1},children:(0,_jsxRuntime.jsx)(_native.NavigationContainer,{children:(0,_jsxRuntime.jsx)(Tab.Navigator,{children:(0,_jsxRuntime.jsx)(Tab.Screen,{name:"Home",component:_HomeScreen.HomeScreen})})})});};function App(){return(0,_jsxRuntime.jsx)(AppNavigator,{});}