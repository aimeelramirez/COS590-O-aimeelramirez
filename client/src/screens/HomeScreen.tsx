/* eslint-disable prettier/prettier */
import React, {useState, useEffect} from 'react';
import {View, Text, TouchableOpacity, StyleSheet} from 'react-native'; // Make sure these imports work for React Native for Windows

const HomeScreen: React.FC = () => {
  return (
    <View style={styles.container}>
      <Text style={{color: 'black'}}>Home Screen</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
    justifyContent: 'center',
  },
  payButton: {
    marginTop: 16,
    padding: 12,
    backgroundColor: 'blue',
    borderRadius: 8,
    alignItems: 'center',
  },
  payButtonText: {
    color: 'white',
    fontWeight: 'bold',
  },
  cancelButton: {
    marginTop: 16,
    padding: 12,
    backgroundColor: 'red',
    borderRadius: 8,
    alignItems: 'center',
  },
  cancelButtonText: {
    color: 'white',
    fontWeight: 'bold',
  },
});

export default HomeScreen;
