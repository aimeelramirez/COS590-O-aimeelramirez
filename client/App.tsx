/* eslint-disable react/no-children-prop */
/* eslint-disable prettier/prettier */
import * as React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import { createMaterialBottomTabNavigator } from 'react-native-paper/react-navigation';
import HomeScreen from './src/screens/HomeScreen'; 
import OnboardingScreen from './src/components/OnboardingScreen'
// import FontAwesome from 'react-native-vector-icons/fontawesome';
import {View} from 'react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';

const Tab = createMaterialBottomTabNavigator();

const AppNavigator = () => {
  return (
    <View style={{backgroundColor: 'white', flex: 1}}>
      <NavigationContainer>
      <Tab.Navigator initialRouteName="OnboardingScreen" > 
        <Tab.Screen name="Onboarding" component={OnboardingScreen}/>
      <Tab.Screen
            name="Home"
            component={HomeScreen}
            // options={{
            //   tabBarIcon: () => (
            //   // <FontAwesome name="home" color="#ff0000" size={20} />
            //   ),
            // }}
          />
        </Tab.Navigator>
      </NavigationContainer>
    </View>
  );
};

const App = () => {
  return <SafeAreaProvider><AppNavigator /></SafeAreaProvider>
};

export default App;
