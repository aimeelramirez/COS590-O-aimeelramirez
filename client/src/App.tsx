/* eslint-disable react/no-children-prop */
/* eslint-disable prettier/prettier */
import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import HomeScreen from './screens/HomeScreen'; // Correct import path
import Icon from 'react-native-vector-icons/FontAwesome'; // Choose the icon set you want to use

import { View } from 'react-native';

const Tab = createBottomTabNavigator();

const AppNavigator: React.FC = () => {
  return (
    <View style={{ backgroundColor: 'pink', flex: 1 }}>
      <NavigationContainer>
        <Tab.Navigator>
          <Tab.Screen
            name="Home"
            component={HomeScreen}
            options={{
              tabBarIcon: ({ color, size }) => (
                // Your custom icon component or an Image
                <Icon name="star" size={30} color="gold" />
              ),
            }}
          />
        </Tab.Navigator>
      </NavigationContainer>
    </View>
  );
};

const App: React.FC = () => {
  return <AppNavigator />;
};

export default App;
