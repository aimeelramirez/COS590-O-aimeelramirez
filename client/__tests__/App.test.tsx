/**
 * @format
 */
jest.useFakeTimers();
import 'react-native';
import React from 'react';
import App from '../src/App';

// Note: import explicitly to use the types shipped with jest.
// import {it} from '@jest/globals';
import '@testing-library/react-native/extend-expect';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import AppNavigator from '../src/App'; // Update the import path accordingly
import HomeScreen from '../src/screens/HomeScreen';


describe('HomeScreen on navigate', () => {
  it('navigates to HomeScreen', async () => {
    const { findByText } = render(<HomeScreen/>);
    const HomeScreenText = findByText('Home Screen');
    expect(HomeScreenText).toBeTruthy();
  });
});
it('renders correctly Navigator', () => {
  const {toJSON} = render(<AppNavigator />);
  expect(toJSON()).toMatchSnapshot();
});
it('renders correctly App', async () => {
  renderer.create(<App />);
});
