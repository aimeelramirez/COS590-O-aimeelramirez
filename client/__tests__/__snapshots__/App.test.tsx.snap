// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Navigator 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
